pkg/apis/authorization/types.go

```diff
@@ -17,13 +17,19 @@ limitations under the License.
 package authorization
 
 import (
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
+// +genclient=true
+// +nonNamespaced=true
+// +noMethods=true
+
 // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
 // spec.namespace means "in all namespaces".
 type SubjectAccessReview struct {
 	unversioned.TypeMeta
+	api.ObjectMeta
```

Comment:
I don't really want to propagate this problem though.  ObjectMeta is not
intended for things that aren't objects.  Code that assumes that is broken,
and we're just going to be piling on to the bad here.

On Thu, Aug 4, 2016 at 3:09 PM, David Eads <notifications@github.com> wrote:

> In pkg/apis/authorization/types.go
> <https://github.com/kubernetes/kubernetes/pull/20573#discussion_r73584170>
> :
>
> >  // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
> >  // spec.namespace means "in all namespaces".
> >  type SubjectAccessReview struct {
> >  	unversioned.TypeMeta
> > +	api.ObjectMeta
>
> What is kubectl create doing that it requires there to be ObjectMeta?
> What would be the minimal changes to kubectl to be able to handle these
> objects without ObjectMeta?
>
> It seems to be required API server side for setting the self-link. For a
> long time we had a special-cased hack in origin to skip that for certain
> types. I don't know how its working downstream now.
>
> Given the difficulty in making it work and the amount of code that assumes
> that every type has ObjectMeta, this seems lower friction.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20573/files/42b8be284e0c00b2658d26d652153ea70c691626#r73584170>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p9_upc1XxX1MamW5YN3XoqZ3rUKLks5qcjjbgaJpZM4HSwZB>
> .
>